### PR TITLE
[Issue #2664] Join data from multiple sprint boards

### LINKED ISSUE #2664: Export and combine data for multiple sprint boards
### Summary

Because Agile Six and Nava use separate projects to track and organize our sprints, we need to be able to export and combine data from multiple sprint boards to accurately report on percent complete for each deliverable in the quad.

### Open questions

If an issue appears in multiple sprint boards, how should we:
- Retain a copy of both sets of metadata (could result in double counting at deliverable level)
- Prioritize one as the official source of metadata

### Acceptance criteria

- [ ] Running a single export command allows users to export and combine data for multiple sprint boards into a  single dataset (i.e. `GitHubIssues`)
- [ ] The data from each sprint board can be separated again using the board name or the GitHub project number.

### PR DESCRIPTION
**Pull request by widal001**
## Summary

Joins data from multiple sprint boards in `make delivery-data-export`

Fixes #2664 

### Time to review: __5 mins__

- Adds a new `analytics.etl` sub-package and moves the transformation code in `analytics.datasets.issues` to `analytics.etl.github`
- Updates the `export_sprint_data()` and `export_roadmap_data()` to include `project_owner` and `project_number` fields
- Updates the `export_github_data()` entry point to accept a config file instead of separate command line options for the sprint board owner, project number, sprint field, etc.
- Allows users to specify multiple sprint project boards in the config file that will be loaded and combined in the `GitHubProjectETL` class
- Updates docs to reference update entrypoint for GitHub export

1. Checkout the PR
2. Follow the instructions to install and run it natively
3. Run `make delivery-data-export`
4. You should see the following files in the gitignored `data/` directory:
   - `roadmap-data.json` Intermediate output file with roadmap data
   - `sprint-data-13.json` Intermediate output file with sprint data from HHS/13 project (Nava)
   - `sprint-data-17.json` Intermediate output file with sprint data from HHS/17 project (P&D)
   - `delivery-data.json` Final output file with data from both sprint boards combined
5. Run `make sprint-burndown` and you should see two burndown charts and summaries get generated.

Here's the output of `make sprint-burndown` (after running `make delivery-data-export`)

Screenshot 2024-11-04 at 2 45 44 PM
Screenshot 2024-11-04 at 2 46 41 PM
Screenshot 2024-11-04 at 2 46 34 PM

### CODE REVIEW

File: analytics/Makefile [hunk 1]
```diff
@@ -7,6 +7,8 @@ REPO ?= simpler-grants-gov
 SPRINT_PROJECT ?= 13
 ROADMAP_PROJECT ?= 12
 OUTPUT_DIR ?= data
+CONFIG_DIR ?= config
+PROJECT_CONFIG_FILE ?= $(CONFIG_DIR)/github-projects.json
```
Comment (widal001) [inline analytics/Makefile:11]: @coilysiren We don't need to turn this on in AWS step functions yet, but when we're ready to turn on the corresponding step functions, is there anything special we need to do to reference a version controlled config file in the same repo as the source application?

IIRC we had to update the path to the mounted volume for the `OUTPUT_DIR,` but I wasn't sure if we needed to something similar for this config file

File: analytics/config/github-projects.json
Comment (widal001) [inline analytics/config/github-projects.json:1]: @coilysiren here's the version controlled config file

**Comment by mdragon** [comment]
Don't see anything I'd flag here, but tagging in @babebe and @mikehgrantsgov  who know Python better than I do.

File: analytics/src/analytics/datasets/issues.py [hunk 2]
```diff
@@ -127,133 +128,12 @@ def get_sprint_name_from_date(self, date: pd.Timestamp) -> str | None:
...
-        lookup = populate_issue_lookup_table(lookup, roadmap_data_in)
-        lookup = populate_issue_lookup_table(lookup, sprint_data_in)
-        # Flatten and write issue level data to output file
-        issues = flatten_issue_data(lookup)
-        return cls(pd.DataFrame(data=issues))
-
-
-# ===============================================================
-# Transformation helper functions
-# ===============================================================
```
Comment (widal001) [inline analytics/src/analytics/datasets/issues.py:157]: All of the code below this was just moved to `analytics/etl/github.py`

File: analytics/src/analytics/etl/github.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+    owner: str
+    project_number: int
+    points_field: str = "Points"
+    sprint_field: str = "Sprint"
+
+
+# ========================================================
+# ETL pipeline
+# ========================================================
```
Comment (widal001) [inline analytics/src/analytics/etl/github.py:90]: All of the code below this was relocated from `/analytics/datasets/issues.py` after creating this new ETL sub-package

File: analytics/tests/datasets/test_issues.py [hunk 4]
```diff
@@ -22,158 +17,6 @@
 )
 
 
-class TestGitHubIssues:
-    """Test the GitHubIssues.load_from_json_files() class method."""
-
```
Comment (widal001) [inline analytics/tests/datasets/test_issues.py:27]: All of these tests were just relocated to `/tests/etl/test_github.py`

File: analytics/tests/etl/test_github.py [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    dataset_wanted = [i.model_dump() for i in output_data]
+    files_wanted = [InputFiles(roadmap=roadmap_file, sprint=sprint_file)]
+    # Act - run the ETL
+    etl.run()
+    # Assert
+    assert etl._transient_files == files_wanted
+    assert etl.dataset.to_dict() == dataset_wanted
+
+
+class TestGetParentWithType:
```
Comment (widal001) [inline analytics/tests/etl/test_github.py:186]: All of these tests were relocated from `/tests/datasets/test_issues.py`

**Review by DavidDudas-Intuitial — APPROVED** [review]
LGTM

**Pull request merged**